### PR TITLE
test: analytics2

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Related Issues
#745 

### Description and Example
This is some tests for analytics2

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NewAnalyticsScreenBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                labelCancel = "cancel",
+                onPressOk = nativeNavigation()
+            )
+        )
+    )
+
+    private fun navigateToPageButton() =  Button(
+        text = "navigateToPage",
+        onPress = listOf(
+            Navigate.PushView(Route.Remote(NEW_ANALYTICS_NAVIGATE_ENDPOINT, true), analytics = ActionAnalyticsConfig.Disabled),
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NewAnalyticsScreenBuilder.kt:111]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/64ab957eafc0de86a742fff99f81a38e9c978831/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NewAnalyticsScreenBuilder.kt#L111"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: tests/appium/app-ios/AppiumApp/AppiumApp/AnalyticsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    func toJsonString() -> String {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        if #available(iOS 13.0, *) {
+            encoder.outputFormatting.insert(.withoutEscapingSlashes)
+        }
+
+        let data = try! encoder.encode(self)
+        return String(data: data, encoding: .utf8)!
```
Comment (botbeaglezup) [inline tests/appium/app-ios/AppiumApp/AppiumApp/AnalyticsViewController.swift:78]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/db20426e417ed6fab3435067215d7e9cc0d63b77/tests/appium/app-ios/AppiumApp/AppiumApp/AnalyticsViewController.swift#L78"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AnalyticsViewController.swift:78</code></td>
    </tr>
  </tbody>
</table>

File: tests/appium/app-ios/AppiumApp/AppiumApp/BeagleConfig/DeepLink/DeepLinkScreenManager.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+import Beagle
+
+class DeepLinkScreenManager: DeepLinkScreenManaging {
+    func getNativeScreen(with path: String, data: [String : String]?) throws -> UIViewController {
```
Comment (botbeaglezup) [inline tests/appium/app-ios/AppiumApp/AppiumApp/BeagleConfig/DeepLink/DeepLinkScreenManager.swift:21]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/db20426e417ed6fab3435067215d7e9cc0d63b77/tests/appium/app-ios/AppiumApp/AppiumApp/BeagleConfig/DeepLink/DeepLinkScreenManager.swift#L21"></span>Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.<br />
<code>colon</code> <code>DeepLinkScreenManager.swift:21</code></td>
    </tr>
  </tbody>
</table>

File: tests/appium/app-android/app/build.gradle [hunk 4]
```diff
@@ -59,7 +59,7 @@ dependencies {
 
     implementation project(Dependencies.Modules.core)
     kapt project(Dependencies.Modules.processor)
-
+    implementation 'com.google.code.gson:gson:2.8.6'
```
Comment (matheusribeirozup) [inline tests/appium/app-android/app/build.gradle:62]: What do you think to put this dependency in `Dependencies` file?

File: tests/appium/project/src/test/resources/features/Analytics.feature [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    Then a dialog should appear on the screen with text Confirm Message
+    When I click on text ACCEPT
+    Then an analytics record should be created for Analytics 02
+
+  Scenario: Analytics 03 - Action with remote analytics config and not declared in the local config (should create record with params from remote config)
+    When I click on text Alert with remote analytics configuration
+    Then a dialog should appear on the screen with text AlertMessage
+    When I click on text OK
+    Then an analytics record should be created for Analytics 03
+
```
Comment (matheusribeirozup) [inline tests/appium/project/src/test/resources/features/Analytics.feature:45]: What do you think to put one more scenario to cover when both local and remote config are declared to verify what should be used?

File: tests/appium/app-android/app/build.gradle (hunk 4 above)
Comment (viniciusguardieirozup) [reply tests/appium/app-android/app/build.gradle]: I think it is a good idea, I'll do this change

File: tests/appium/project/src/test/resources/features/Analytics.feature (hunk 5 above)
Comment (Tiagoperes) [reply tests/appium/project/src/test/resources/features/Analytics.feature:45]: I'm not sure if we should worry about covering every case in these e2e tests. I'd say this is type of coverage we should have with unit tests. Here, I think it suffices to have a test that use the remote config and another that uses the local config. just to see if both can be accessed.

**Review by matheusribeirozup — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**